Template crash on individual post pages (Chorus mode): can't evaluate field UserPage in type CollectionPostPage

## Description

On a multi-user (Chorus) instance, viewing an individual published post causes a template execution error that cuts off the rest of the page render. The post content and footer never render for visitors.

## Steps to reproduce

1. Run a multi-user WriteFreely instance (v0.17.1, Docker image `ghcr.io/writefreely/writefreely:latest`)
2. Publish a post to any collection/blog (e.g. `/{collection}/{slug}`)
3. Visit the individual post page as an anonymous visitor

## Observed error (server log)

```
ERROR: posts.go:1681: Error in chorus-collection-post template: template: header.tmpl:33:18: executing "user-navigation" at <.UserPage.Username>: can't evaluate field UserPage in type writefreely.CollectionPostPage
```

## Impact

The HTTP response still returns `200`, but the HTML is truncated mid-render inside `<header>` / `<nav id="user-nav">`. The article body, footer, and closing tags never get output. Confirmed via `curl` that the response ends abruptly right after the Cloudflare Rocket Loader script tag inside the header markup.

## Environment

- WriteFreely v0.17.1 (Docker, `ghcr.io/writefreely/writefreely:latest`)
- Multi-user / Chorus mode enabled
- SQLite database

--- a/collections.go
+++ b/collections.go
@@ -414,6 +414,18 @@ func (c CollectionPage) DisplayMonetization() string {
 	return displayMonetization(c.Monetization, c.Alias)
 }
 
+// UserPage provides the fields expected by the shared "user-navigation"
+// template, which otherwise assumes it's rendering for a page that embeds
+// *UserPage (e.g. the "me" backend pages).
+func (c CollectionPage) UserPage() *UserPage {
+	return &UserPage{
+		StaticPage: c.StaticPage,
+		IsAdmin:    c.IsAdmin,
+		CanInvite:  c.CanInvite,
+		CollAlias:  c.CollAlias,
+	}
+}
+
 func (c *DisplayCollection) Direction() string {
 	if c.Language == "" {
 		return "auto"

--- a/posts.go
+++ b/posts.go
@@ -303,6 +303,18 @@ func (p *Post) HasTitleLink() bool {
 	return hasLink
 }
 
+// UserPage provides the fields expected by the shared "user-navigation"
+// template, which otherwise assumes it's rendering for a page that embeds
+// *UserPage (e.g. the "me" backend pages).
+func (c CollectionPostPage) UserPage() *UserPage {
+	return &UserPage{
+		StaticPage: c.StaticPage,
+		IsAdmin:    c.IsAdmin,
+		CanInvite:  c.CanInvite,
+		CollAlias:  c.CollAlias,
+	}
+}
+
 func (c CollectionPostPage) DisplayMonetization() string {
 	if c.Collection == nil {
 		log.Info("CollectionPostPage.DisplayMonetization: c.Collection is nil")

